### PR TITLE
Restructure files into 7 levels to provide a better understanding/sense of progression.

### PR DESCRIPTION
**Pull request by wjlow**
- This makes it easier for people doing the exercise in their own time to know the progression of the exercises.
- I hope it provides a better sense of accomplishment and progression
- Should make teaching easier
- This removes dependencies between packages

### CODE REVIEW

File: src/main/scala/introcourse/level04/package.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package introcourse
+
+package object level04 {
+
+  case class Person(name: String, age: Int)
```
Comment (felipeeflores) [inline src/main/scala/introcourse/level04/package.scala]: let's make case classes `final` 🤓

File: src/main/scala/introcourse/level05/EitherExercises.scala [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-import fundamentals.level03.ExceptionExercises.personStringPairs
+import ExceptionExercises.personStringPairs
 
 /**
   * These exercises show the alternative to throwing Exceptions using the `Either` data type.
   * We will treat errors as values that our functions return.
   */
-object Exceptions2EitherExercises {
+object EitherExercises {
```
Comment (felipeeflores) [inline src/main/scala/introcourse/level05/EitherExercises.scala]: 👍

File: src/main/scala/introcourse/level05/package.scala [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package introcourse
+
+package object level05 {
+
+  case class Person(name: String, age: Int)
```
Comment (felipeeflores) [inline src/main/scala/introcourse/level05/package.scala]: `final`

File: src/main/scala/introcourse/level01/IntroExercises.scala [hunk 4]
```diff
@@ -1,5 +1,4 @@
-package fundamentals.level01
-
+package introcourse.level01
```
Comment (felipeeflores) [inline src/main/scala/introcourse/level01/IntroExercises.scala]: I like this name better 👍

File: src/main/scala/introcourse/level04/package.scala (hunk 1 above)
Comment (wjlow) [reply src/main/scala/introcourse/level04/package.scala]: This is one of the good things that we won't teach because we don't need the distraction.

Comment (felipeeflores) [reply src/main/scala/introcourse/level04/package.scala]: I don't think you need to explain anything about it. It's like trying to explain why a class in java can be annotated with `final`. They can look it up or we answer if anyone asks.

Comment (felipeeflores) [reply src/main/scala/introcourse/level04/package.scala]: Additionally, it sets the student in the right path (copy/paste pattern) and the one that knows and understands doesn't see this as miss or anti-pattern.

Having said that the request is irrelevant for this PR and I'm not so fussed about it. So feel free to merge at will.

**Pull request merged**